docs/core/diagnostics/dotnet-dump.md

```diff
@@ -0,0 +1,221 @@
+ms.author: stmaclea
+ms.date: 10/01/2019
+---
+# Dump collection and analysis utility (`dotnet-dump`)
+
+**This article applies to: ✓** .NET Core 3.0 SDK and later versions
+
+> [!NOTE]
+> `dotnet-dump` isn't supported on macOS.
+
+## Installing `dotnet-dump`
+
+To install the latest release version of the `dotnet-dump` [NuGet package](https://www.nuget.org/packages/dotnet-dump), use the [dotnet tool install](../tools/dotnet-tool-install.md) command:
+
+```dotnetcli
+dotnet tool install -g dotnet-dump
+```
+
+## Synopsis
+
+```console
+dotnet-dump [-h|--help] [--version] <command>
+```
+
+## Description
+
+The `dotnet-dump` global tool is a way to collect and analyze Windows and Linux dumps without any native debugger involved like `lldb` on Linux. This tool is important on platforms like Alpine Linux where a fully working `lldb` isn't available. The `dotnet-dump` tool allows you to run SOS commands to analyze crashes and the garbage collector (GC), but it isn't a native debugger so things like displaying native stack frames aren't supported.
+
+## Options
+
+- **`--version`**
+
+  Displays the version of the dotnet-counters utility.
+
+- **`-h|--help`**
+
+  Shows command-line help.
+
+## Commands
+
+| Command                                     |
+| ------------------------------------------- |
+| [dotnet-dump collect](#dotnet-dump-collect) |
+| [dotnet-dump analyze](#dotnet-dump-analyze) |
+
+## dotnet-dump collect
+
+Captures a dump from a process.
+
+### Synopsis
+
+```console
+dotnet-dump collect [-h|--help] [-p|--process-id] [--type] [-o|--output] [--diag]
+```
+
+### Options
+
+- **`-h|--help`**
+
+  Shows command-line help.
+
+- **`-p|--process-id <PID>`**
+
+  Specifies the process ID number to collect a memory dump from.
+
+- **`--type <Heap|Mini>`**
+
+  Specifies the dumb type, which determines the kinds of information that are collected from the process. There are two types:
+
+    - `Heap` - A large and relatively comprehensive dump containing module lists, thread lists, all stacks, exception information, handle information, and all memory except for mapped images.
+    - `Mini` - A small dump containing module lists, thread lists, exception information, and all stacks.
+
+  If not specified, `Heap` is the default.
+
+- **`-o|--output <output_dump_path>`**
+
+  The full path and file name where the collected dump should be written.
+
+  If not specified:
+
+    - Defaults to *.\dump_YYYYMMDD_HHMMSS.dmp* on Windows.
+    - Defaults to *./core_YYYYMMDD_HHMMSS* on Linux.
+
+    YYYYMMDD is Year/Month/Day and HHMMSS is Hour/Minute/Second.
+
+- **`--diag`**
+
+  Enables dump collection diagnostic logging.
+
+## dotnet-dump analyze
+
+Starts an interactive shell to explore a dump. The shell accepts various [SOS commands](#analyze-sos-commands).
+ 
+### Synopsis 
+
+```console
+dotnet-dump analyze [-h|--help] <dump_path> [-c|--command]
+```
+
+### Arguments
+
+- **`<dump_path>`**
+
+  Specifies the path to the dump file to analyze.
```

Comment:
Question: is this required?

Reply:
It is a required argument. I don't think there is a default.

Reply:
So it looks good as-is. Was just checking.